app/bin/service/search.dart

```diff
@@ -94,27 +94,33 @@ Future _main(FrontendEntryMessage message) async {
     _updateDartSdkIndex().whenComplete(() {});
 
     final BatchIndexUpdater batchIndexUpdater = BatchIndexUpdater();
-    await batchIndexUpdater.initSnapshot();
-
-    final scheduler = TaskScheduler(
-      batchIndexUpdater,
-      [
-        ManualTriggerTaskSource(taskReceivePort.cast<Task>()),
-        IndexUpdateTaskSource(db.dbService, batchIndexUpdater),
-        DatastoreHeadTaskSource(
-          db.dbService,
-          TaskSourceModel.scorecard,
-          sleep: const Duration(minutes: 10),
-          skipHistory: true,
-        ),
-        batchIndexUpdater.periodicUpdateTaskSource,
-      ],
+    // Don't block on init, we need to serve liveliness and readiness checks.
+    batchIndexUpdater.init().then(
+      (_) {
+        final scheduler = TaskScheduler(
+          batchIndexUpdater,
+          [
+            ManualTriggerTaskSource(taskReceivePort.cast<Task>()),
+            IndexUpdateTaskSource(db.dbService, batchIndexUpdater),
+            DatastoreHeadTaskSource(
+              db.dbService,
+              TaskSourceModel.scorecard,
+              sleep: const Duration(minutes: 10),
+              skipHistory: true,
+            ),
+            batchIndexUpdater.periodicUpdateTaskSource,
+          ],
+        );
+        scheduler.run();
+
+        Timer.periodic(const Duration(minutes: 5), (_) {
+          updateLatestStats(scheduler.stats());
+        });
+      },
+      onError: (e, StackTrace st) {
+        _logger.shout('Error initializing search service.', e, st);
```

Comment:
If this fails, then what?

Don't we need to crash the service? Exit non-zero, or something? We could have retries, but assuming all the fragile I/O is already wrapped in retry statements, maybe it's best to just give up at this point.

Otherwise, the service will keep being alive. And have a liveness check that returns OK.

Reply:
We could probably just call exit...